torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -73,6 +73,9 @@ Tensor toNonOptFwGrad(const c10::optional<Tensor>& t) {
 }
 
 Tensor toNonOptPrimal(const c10::optional<Tensor>& t) {
+  // if (t.has_value()) {
+  //   std::cout << "toNonOptPrimal: " << std::endl;
+  // }
```

Comment:
@guilhermeleobas the fix is probably:
- toNonOptPrimal should just return t if t is a wrapped number
- when functorch creates a TensorWrapper, if the thing it is wrapping is a wrapped number, then the TensorWrapper should also be a wrapped number

Reply:
Just pushed your suggestion.